### PR TITLE
feat(router): add relative paths handling. resolves #2

### LINKED ISSUE #2: [Feat] Add relative path support to linkTo

### PR DESCRIPTION
**Pull request by meeroslav**

### CODE REVIEW

File: libs/angular-routing/src/lib/url-parser.ts [hunk 1]
```diff
@@ -5,4 +5,24 @@ export class UrlParser {
...
+
+  joinUrls(currentUrl: string, url: string): string {
+    const currentUrlSegments = currentUrl
+      .split('#')[0] // remove hash
+      .split('?')[0] // remove query params
+      .split('/');
+    const urlSegments = url.split('/');
+
+    return urlSegments
+      .reduce((segments, segment) => {
```
Comment (kyleryanbanks) [inline libs/angular-routing/src/lib/url-parser.ts:17]: I wouldn't have considered using a reducer like this.

Makes for a really clean transform.

Comment (meeroslav) [reply libs/angular-routing/src/lib/url-parser.ts:17]: Hopefully, it's also easy to read.

File: libs/angular-routing/src/lib/router.service.ts [hunk 2]
```diff
@@ -47,6 +47,10 @@ export class Router {
   }
 
   serializeUrl(url: string, queryParams?: Params, hash?: string) {
+    // if relative path
+    if (!url.startsWith('/')) {
+      url = this.urlParser.joinUrls(this.location.path(), url);
```
Comment (brandonroberts) [inline libs/angular-routing/src/lib/router.service.ts:52]: The URL Web API supports building relative URLs, so we can leverage that. WDYT?

https://developer.mozilla.org/en-US/docs/Web/API/URL/URL

```suggestion
      url = this.urlParser.parse(url, this.getLocation()).pathname;
```

Comment (meeroslav) [reply libs/angular-routing/src/lib/router.service.ts:52]: I've looked into that solution first. There is one fine difference to the current implementation:
- The `joinUrls` keeps query params and hash from the target URL making functions like `serializeUrl` (and `go`) work without the need of explicit path and params separation

Let me know what you think.

Comment (brandonroberts) [reply libs/angular-routing/src/lib/router.service.ts:52]: I tested it with passing query params and a hash, and it parsed it correctly. That being said, it provides a little less overhead if you want to use it separately.

**Pull request merged**